### PR TITLE
Bugfix and maintenance rawprepare

### PR DESCRIPTION
**Pull request by jenshannoschwalm**
There were a number of issues (at least #13535 #13537) with reports of dt crashing, all given logs show something very weird later leading to a crash.
There seemed to be something about a) chosen demosaicer or b) some tiling issue but this is all related to an earlier problem.

Somehow - the reason for **how** this is happening not yet understood - we have just plain bad crop parameters while `commit_params()` for rawprepare.

Not fixed would be img->width or img->height **not** being correct at this stage

This PR implements a check for meaningful crop values and if not correct
1. resets to something "safe" for processing
2. give a trouble warning and output to log

to make dt at least not crashing.

While being here i renamed the elements in parameters and gtkwidgets to the "what does it mean" to allow better code reading & understanding.

1. x -> left
2. y -> top
3. width ->right
4. height -> bottom

### CODE REVIEW

**Comment by TurboGit** [comment]
> we have just plain bad crop parameters while `commit_params()` for rawprepare.

If you have a backtrace for a call to commit_params() with bad parameters it would be nice. This is really worrisome, especially since few people are reproducing.

On of the possible culprit could be the "recent" added support for gainmap? See commit 9552de237e70b6811214950bc8c30b6f9cce3d1c. Maybe some images have broken gainmap data which are playing a bad game with crop values? Just very wild guess but rawprepare has not received lot of changes those last years and this issue seems recent.

**Comment by jenshannoschwalm** [comment]
No, unfortunately I don't have one.

BUT,the logs presented in mentioned issues allow only one of two Analyse

1. there is no valid image size in img structure
2. crop parameters have been bad. This could be of many reasons, style, copy&paste of history,your new preset 😃 

You may see a new FIXME at the gainmaps section, I don't understand that part yet. At least we might get more valuable logs after this and users might be able to tell what they did.

BTW, do see a chance to write existing presets to the log in a meaningful way?

**Review by TurboGit — APPROVED** [review]
Looks good to me, thanks!

**Comment by TurboGit** [comment]
I had a look (did a reformatting too) and think the following piece of code is wrong:

```
    // covers the entire image, and is not a 1x1 no-op
    dt_dng_gain_map_t *g = (dt_dng_gain_map_t *)g_list_nth_data(image->dng_gain_maps, i);
    if(g == NULL
       || g->plane != 0
       || g->planes != 1
       || g->map_planes != 1
       || g->row_pitch != 2
       || g->col_pitch != 2
       || g->map_points_v < 2
       || g->map_points_h < 2
       || g->top > 1
       || g->left > 1
       || g->bottom != image->height
       || g->right != image->width)
      return FALSE;
```

It checks that it covers the entire image, so shouldn't be the last 4 lines:

```
       || g->top != 0
       || g->left != 0
       || g->bottom != (image->height - 1)
       || g->right != (image->width - 1)
```

File: src/iop/rawprepare.c [hunk 1]
```diff
@@ -574,20 +606,44 @@ static int image_is_normalized(const dt_image_t *const image)
...
+        int right,
+        int top,
+        int bottom)
 {
   dt_image_t *img = NULL;
   img = dt_image_cache_get(darktable.image_cache, imgid, 'r');
-  const gboolean test = (img->p_width == img->width - dx)
-                     && (img->p_height == img->height - dy);
+
+  const gboolean cropvalid = (left >= 1) && (right >= 0) && (top >= 0) && (bottom >= 0)
```
Comment (TurboGit) [inline src/iop/rawprepare.c:620]: Just spotted this now, `left >= 0` , no?

Comment (jenshannoschwalm) [reply src/iop/rawprepare.c:620]: Absolutely. Had this while testing for the trouble message and forgot.

I am on the bike, feel free to fix.

Btw, one more idea while riding, we have the libraw open now, some people were overriding the cr3 by renaming. All libraw files might introduce bugs...

Comment (TurboGit) [reply src/iop/rawprepare.c:620]: > I am on the bike, feel free to fix.

Fixed! So you are also a Cyclist :)

**Pull request merged**